osh/word_parse.py

```diff
@@ -1130,9 +1130,12 @@ def ParseCommandExpr(self):
         """
         enode, last_token = self.parse_ctx.ParseYshExpr(self.lexer,
                                                         grammar_nt.command_expr)
-        if last_token.id == Id.Op_RBrace:
-            last_token.id = Id.Lit_RBrace
-        self.buffered_word = last_token
+
+        if last_token.id == Id.Op_RBrace or last_token.id == Id.Op_Semi:
```

Comment:
I would add a comment here -- what invariant are we trying to preserve?  I guess it's tested in the unit tests, but it's nice to have a description too

It looks like this is 2 of 4 cases in end_stmt -- } and ;

The other 2 are Op_Newline and Eof_Real

(I wonder if we need those as buffered words -- maybe we can Unread there too ?)